pkg/handler/repositories.go

```diff
@@ -222,3 +275,91 @@ func (rh *RepositoryHandler) deleteRepository(c echo.Context) error {
 	}
 	return c.NoContent(http.StatusNoContent)
 }
+
+//
+// Helper methods
+//
+
+func (rh *RepositoryHandler) mapRepositoryResponse2EventMessage(input *api.RepositoryResponse) (*message.IntrospectRequestMessage, error) {
+	if input == nil {
+		return nil, fmt.Errorf("input cannot be nil")
+	}
+	output := &message.IntrospectRequestMessage{
+		Uuid: input.UUID,
+		Url:  input.URL,
+	}
+	return output, nil
+}
+
+func (rh *RepositoryHandler) produceMessage(c echo.Context, msg *message.IntrospectRequestMessage) error {
+	var (
+		headerKey string
+		err       error
+	)
+	// Fill Key
+	key := msg.Uuid
+
+	// Read header values
+	headerKey = string(message.HdrXRhIdentity)
+	xrhIdentity := GetHeader(c, headerKey, []string{})
+	if len(xrhIdentity) == 0 {
+		return fmt.Errorf("expected a value for '%s' http header", headerKey)
+	}
+	headerKey = string(message.HdrXRhInsightsRequestId)
+	xrhInsightsRequestId := GetHeader(c, headerKey, []string{})
+	if len(xrhInsightsRequestId) == 0 {
+		return fmt.Errorf("expected a value for '%s' http header", headerKey)
+	}
+
+	// Fill headers
+	headers := []kafka.Header{
+		{
+			Key:   string(message.HdrType),
+			Value: []byte(message.HdrTypeIntrospect),
+		},
+		{
+			Key:   string(message.HdrXRhIdentity),
+			Value: []byte(xrhIdentity[0]),
+		},
+		{
+			Key:   string(message.HdrXRhInsightsRequestId),
+			Value: []byte(xrhInsightsRequestId[0]),
+		},
+	}
+
+	// Fill topic
+	topic := schema.TopicIntrospect
+
+	// Produce message
+	if err = event.Produce(rh.Producer, topic, key, msg, headers...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (rh *RepositoryHandler) mapRepositoryBulkCreateResponse2EventMessage(input *api.RepositoryBulkCreateResponse) (*message.IntrospectRequestMessage, error) {
+	if input == nil {
+		return nil, fmt.Errorf("input cannot be nil")
+	}
+	output := &message.IntrospectRequestMessage{
+		Uuid: input.Repository.UUID,
+		Url:  input.Repository.URL,
+	}
+	return output, nil
+}
+
+func (rh *RepositoryHandler) bulkProduceMessages(c echo.Context, response []api.RepositoryBulkCreateResponse) error {
```

Comment:
Similar to my previous comment, I wonder if the kafka related methods in here should be in their own package. Then from here we just called something like `event.ProduceIntrospectMessage(msg)`, similar to how we interact with the dao layer.

Reply:
Yep! I agree! I don't like the place where I put that code either! I have to move to `pkg/event`

Reply:
moved some logic to pkg/event/producer and pkg/event/adapter; I have to update pkg/event/README.md to provide more documentation about this; in short:
- 1 interface per specific message producer; all of them use the common code for a generic message.
- The interface is registered with the http handler that will use it.
- From the handler the specific message producer is available and it is called previous call to the adapter to translate to the message format.